src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -308,17 +308,24 @@ public void Shuffle<T>(T[] values)
         /// </remarks>
         public void Shuffle<T>(Span<T> values)
         {
-            int n = values.Length;
+            Shuffle(ref MemoryMarshal.GetReference(values), values.Length);
```

Comment:
Thanks, but I don't think we need any of the below changes. I'd just change L297 above to construct the span without AsSpan, e.g.
```C#
Shuffle(new Span<T>(ref MemoryMarshal.GetArrayDataReference(values), values.Length));
```
with a comment indicating why it's safe (because Shuffle will only write into the array things already in the array).

Reply:
Ah, haven't realised this would work too.